app/controllers/auth_source_ldaps_controller.rb

```diff
@@ -18,6 +18,22 @@ def create
     end
   end
 
+  def test_connection
+    result = {}
+    begin
+      temp_auth_source_ldap = AuthSourceLdap.new(params[:auth_source_ldap])
+      temp_auth_source_ldap.ldap_con.valid_user?("")
```

Comment:
For some reason this is hanging on my computer forever, no matter the LDAP backend I choose. Have you ever encountered this? Maybe you want to put a timeout (10sec?)